resources/js/features/user/settings/components/KeysSectionCard/ManageWebApiKey.tsx

```diff
@@ -0,0 +1,110 @@
+import { useMutation } from '@tanstack/react-query';
+import type { AxiosResponse } from 'axios';
+import axios from 'axios';
+import { type FC, useState } from 'react';
+import { LuAlertCircle, LuCopy } from 'react-icons/lu';
+import { useCopyToClipboard, useMedia } from 'react-use';
+
+import { BaseButton } from '@/common/components/+vendor/BaseButton';
+import { toastMessage } from '@/common/components/+vendor/BaseToaster';
+import { SimpleTooltip } from '@/common/components/SimpleTooltip';
+
+import { usePageProps } from '../../hooks/usePageProps';
+
+export const ManageWebApiKey: FC = () => {
+  const { userSettings } = usePageProps<App.Community.Data.UserSettingsPageProps>();
+
+  const [, copyToClipboard] = useCopyToClipboard();
+
+  // Hide the copy button's tooltip on mobile.
+  const isXs = useMedia('(max-width: 640px)', true);
+
+  const [currentWebApiKey, setCurrentWebApiKey] = useState(userSettings.apiKey ?? '');
+
+  const mutation = useMutation({
+    mutationFn: () => {
+      return axios.delete<unknown, AxiosResponse<{ newKey: string }>>(
+        route('settings.keys.web.destroy'),
+      );
+    },
+    onSuccess: ({ data }) => {
+      setCurrentWebApiKey(data.newKey);
+    },
+  });
+
+  const handleCopyApiKeyClick = () => {
+    copyToClipboard(currentWebApiKey);
+    toastMessage.success('Copied your web API key!');
+  };
+
+  const handleResetApiKeyClick = () => {
+    if (!confirm('Are you sure you want to reset your web API key? This cannot be reversed.')) {
+      return;
+    }
+
+    toastMessage.promise(mutation.mutateAsync(), {
+      loading: 'Resetting...',
+      success: 'Your web API key has been reset.',
+      error: 'Something went wrong.',
+    });
+  };
+
+  return (
+    <div className="@container">
+      <div className="flex w-full flex-col @lg:grid @lg:grid-cols-4">
+        <p className="w-48 text-menu-link">Web API Key</p>
+
+        <div className="col-span-3 flex w-full flex-col gap-2">
+          <SimpleTooltip
+            isOpen={isXs ? false : undefined}
+            isWrappingTapTarget={true}
+            tooltipContent="Copy to clipboard"
+          >
+            <BaseButton className="flex gap-2" onClick={handleCopyApiKeyClick}>
+              <LuCopy />
+              <span className="font-mono">{safeFormatApiKey(currentWebApiKey)}</span>
```

Comment:
I'm guessing the truncation is to obscure sensitive information, but it looks weird to see the value collapsed when there's so much room.

![image](https://github.com/user-attachments/assets/f09a68ee-cf61-4116-9c31-7374f893c56e)

Reply:
In latest, this now only goes full-width on small device sizes.

![Screenshot 2024-09-02 at 4 10 01 PM](https://github.com/user-attachments/assets/8efb5fa5-5fcd-4247-ad1b-c3a597e301e8)